### DISCUSSION TITLE
Frigate on Debian, Docker, AMD & Radeon - how to optimize?

### DISCUSSION THREAD

**Discussion by TheStigh** [blakeblackshear/frigate#4664] [Ask A Question]
Hi all,

I have my Frigate installed on a standalone Debian server where Frigate are installed as a docker container using docker-compose to run it. My server are running AMD Ryzen 5 5600G with Radeon Graphics. I've got 2 * Coral DualTPU's running also. Frigate running are 0.11.1
Camera config are 6; 1 * 8K, 2 * 4K and 3 * 2K. Inference shown in Frigate are between 6 and 7 on each detector. 

Currently I'm running following hw acc arguments:
```
ffmpeg:
  global_args: -hide_banner -loglevel fatal
  hwaccel_args: -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format yuv420p

```

I'm not experienced enough to measure the performance except watching Glances, where CPU are around 7% and Mem are 28%.

I want to try to optimize my hw acceleration but are not experienced enough to place the `LIBVA_DRIVER_NAME` correct. Should it be inside the docker-compose file or in the frigate.yml?

And what else can I try to optimize everything?

Please provide an example so I get this right :)

**Comment by NickM-27** [comment]
If your hwaccel is not failing and there are no errors in the logs then the LIBVA_DRIVER_NAME is correct. In the next version of frigate the "debug" page (now renamed to system) will show if your GPU is working & how hard it is working.

**Comment by TheStigh** [comment]
@NickM-27 

Thanks for the quick answer :) Ok, I assume since my CPU are more or less idling < 10% with 6 cameras, also running Home Assistant in a different container, I would guess it works ok.

What is the ETA of that next version?

Finally, could you still please show me how to change the LIBVA_DRIVER_NAME ? I do have another machine running on a 2.nd location, with a 11th Gen Intel. I would love to play with this. Setup are the same except it is Ubuntu and not Debian.

**Reply by NickM-27** [reply]
You can follow the progress of the release here: https://github.com/blakeblackshear/frigate/pull/4055

And in your docker compose you just set

```yaml
frigate:
...
...
  environment:
    LIBVA_DRIVER_NAME: specific_value
```

**Reply by TheStigh** [reply]
Thanx @NickM-27 🥇

**Reply by TheStigh** [reply]
May I ask - what values are supported for the LIBVA_DRIVER_NAME ?

**Reply by NickM-27** [reply]
i965 (older intel)
radeonsi (AMD)
iHD (default value, newer intel)

**Reply by TheStigh** [reply]
what about `radeonsi` ?
And what for newer Intel's?

**Reply by NickM-27** [reply]
Sorry my bad I meant radeonsi instead of vaapi and like I said iHD is for newer intel

**Reply by TheStigh** [reply]
Hehe, my bad also! I read neVer, not neWer ...

Thx!

**Reply by TheStigh** [reply]
How can I see what driver are being used when I'm running Frigate on docker outside HA ?
And how can I see the GPU load?

**Reply by NickM-27** [reply]
In the current version you need to install and run `vainfo` inside the container. Viewing GPU usage depends on what GPU you have but it's either `nvidia-smi` `intel_gpu_top` or `radeontop`

All of this is going to be built in for 0.12 system page

**Reply by NickM-27** [reply]
`apt update && apt install vainfo && vainfo`

**Reply by TheStigh** [reply]
I actually figured out seconds before your reply :)

**Reply by TheStigh** [reply]
Installed vainfo & radeontop, this is the result. 'Graphics pipe' peaks at 14% and drops to 5%. Is this more than acceptable load?

PS; learning a lot now, really appreciate your replies :)

<img width="913" alt="image" src="https://user-images.githubusercontent.com/43895442/206916254-814c6670-e481-4d16-a7e7-26f8ba121bc7.png">

**Reply by NickM-27** [reply]
Looks like expected for the cameras you described

**Reply by TheStigh** [reply]
So this is "good" and what I can expect without a lot more tweaking? Would you be happy with this result?

**Reply by NickM-27** [reply]
I have a 5700G in my server and I got similar results with I used it for all my cams, was happy with it while I used it
